x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -715,31 +743,66 @@ class AgentPolicyService {
       return;
     }
 
-    const policy = await agentPolicyService.get(soClient, agentPolicyId);
-    const fullPolicy = await agentPolicyService.getFullAgentPolicy(soClient, agentPolicyId);
-    if (!policy || !fullPolicy || !fullPolicy.revision) {
-      return;
-    }
+    const policies = await agentPolicyService.getByIDsAsMap(soClient, agentPolicyIds);
+    const fullPolicies = await Promise.all(
+      agentPolicyIds.map((agentPolicyId) =>
+        // TODO: make this a bulk
+        agentPolicyService.getFullAgentPolicy(soClient, agentPolicyId)
+      )
+    );
 
-    const fleetServerPolicy: FleetServerPolicy = {
-      '@timestamp': new Date().toISOString(),
-      revision_idx: fullPolicy.revision,
-      coordinator_idx: 0,
-      data: fullPolicy as unknown as FleetServerPolicy['data'],
-      policy_id: fullPolicy.id,
-      default_fleet_server: policy.is_default_fleet_server === true,
-    };
+    const fleetServerPolicies = fullPolicies.reduce((acc, fullPolicy) => {
+      if (!fullPolicy || !fullPolicy.revision) {
+        return acc;
+      }
 
-    if (policy.unenroll_timeout) {
-      fleetServerPolicy.unenroll_timeout = policy.unenroll_timeout;
-    }
+      const policy = policies[fullPolicy.id];
+      if (!policy) {
+        return acc;
+      }
+
+      const fleetServerPolicy: FleetServerPolicy = {
+        '@timestamp': new Date().toISOString(),
+        revision_idx: fullPolicy.revision,
+        coordinator_idx: 0,
+        data: fullPolicy as unknown as FleetServerPolicy['data'],
+        policy_id: fullPolicy.id,
+        default_fleet_server: policy.is_default_fleet_server === true,
+      };
 
-    await esClient.create({
+      if (policy.unenroll_timeout) {
+        fleetServerPolicy.unenroll_timeout = policy.unenroll_timeout;
+      }
+
+      return [...acc, fleetServerPolicy];
+    }, [] as FleetServerPolicy[]);
+
+    const fleetServerPoliciesBulkBody = fleetServerPolicies.flatMap((fleetServerPolicy) => [
+      { index: { _id: uuid() } },
```

Comment:
Since this migration could potentially be executed more than once by different Kibana nodes, I think we should use a deterministic ID here to avoid creating duplicate documents in `.fleet-policies`.

In other places in Kibana, we've used `uuid.v5()` with an appropriate seed for this. IMO the `policy_id` and `revision_idx` fields should be sufficient, something like this:

```ts
import uuidv5 from 'uuid/v5';

// ...

{ index: { _id: uuidv5(`${fleetServerPolicy.policy_id}:${fleetServerPolicy.revision_idx}`, uuidv5.DNS) } }
```

We'll then need to handle the error on the `esClient.bulk` correctly (and maybe log a debug message?)